### PR TITLE
[exporter/signalfx] Add an option to control the dimension client timeout

### PR DESCRIPTION
**Pull request by dloucasfx**
**Description:** <Describe what has changed.>
The dimension client timeout is currently hardcoded to 10 seconds, this PR makes the timeout configurable.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>
Unit test
**Documentation:** <Describe the documentation added.>
Update Readme

### CODE REVIEW

File: .chloggen/signalfx-exporter-add-dim-timeout.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: signalfxexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add an option to control the dimension client timeout
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (songy23) [inline .chloggen/signalfx-exporter-add-dim-timeout.yaml:13]:
```suggestion
issues: [27815]
```

**Review by songy23 — APPROVED** [review]

File: .chloggen/signalfx-exporter-add-dim-timeout.yaml (hunk 1 above)
Comment (crobert-1) [inline .chloggen/signalfx-exporter-add-dim-timeout.yaml:13]:
```suggestion
issues: [27815]
```

**Review by crobert-1 — COMMENTED** [review]
Please add the option to the README as well

File: .chloggen/signalfx-exporter-add-dim-timeout.yaml (hunk 1 above)
Comment (dloucasfx) [reply .chloggen/signalfx-exporter-add-dim-timeout.yaml]: Thanks; yeah, I tend to open the PR and before updating the change log.
All done, thanks!

**Review by crobert-1 — APPROVED** [review]

**Comment by atoulme** [comment]
Addressing failing build in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27869

**Review by atoulme — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**